### ISSUE TITLE
ddtrace 1.13.0 breaks sqlalchemy

### ISSUE THREAD

**Issue by ollz272** [DataDog/dd-trace-py#5823]
### Summary of problem
We upgraded to ddtrace 1.13.0 and are now getting the following error with fastapi + sqlalchemy:
```
builtins.ValueError: dictionary update sequence element #6 has length 1; 2 is required
Traceback (most recent call last):
  File "/venv/lib/python3.11/site-packages/ddtrace/contrib/asgi/middleware.py", line 255, in __call__
    return await self.app(scope, receive, wrapped_send)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "/venv/lib/python3.11/site-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/venv/lib/python3.11/site-packages/opentelemetry/instrumentation/asgi/__init__.py", line 459, in __call__
    await self.app(scope, otel_receive, otel_send)
  File "/venv/lib/python3.11/site-packages/starlette/middleware/cors.py", line 84, in __call__
    await self.app(scope, receive, send)
  File "/venv/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
    raise exc
  File "/venv/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
  File "/venv/lib/python3.11/site-packages/fastapi/middleware/asyncexitstack.py", line 21, in __call__
    raise e
  File "/venv/lib/python3.11/site-packages/fastapi/middleware/asyncexitstack.py", line 18, in __call__
    await self.app(scope, receive, send)
  File "/venv/lib/python3.11/site-packages/starlette/routing.py", line 718, in __call__
    await route.handle(scope, receive, send)
  File "/venv/lib/python3.11/site-packages/starlette/routing.py", line 276, in handle
    await self.app(scope, receive, send)
  File "/venv/lib/python3.11/site-packages/starlette/routing.py", line 66, in app
    response = await func(request)
               ^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/fastapi/routing.py", line 237, in app
    raw_response = await run_endpoint_function(
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/fastapi/routing.py", line 165, in run_endpoint_function
    return await run_in_threadpool(dependant.call, **values)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/starlette/concurrency.py", line 41, in run_in_threadpool
    return await anyio.to_thread.run_sync(func, *args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/anyio/to_thread.py", line 31, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/anyio/_backends/_asyncio.py", line 937, in run_sync_in_worker_thread
    return await future
           ^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/anyio/_backends/_asyncio.py", line 867, in run
    result = context.run(func, *args)
             ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/timescale_api/routers/site.py", line 68, in list_site
    sites = self.user.all_sites(session)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/timescale_api/auth/user.py", line 185, in all_sites
    return query.all()
           ^^^^^^^^^^^
ValueError: dictionary update sequence element #6 has length 1; 2 is required
```
### Which version of dd-trace-py are you using?
1.13.0
### Which version of pip are you using?
22.3.1

### Which libraries and their versions are you using?

<details>
  <summary>`pip freeze`</summary>
 ` ``
anyio==3.6.2
appdirs==1.4.4
asgiref==3.6.0
assertpy==1.1
astor==0.8.1
astroid==2.15.4
attrs==22.2.0
autoflake==2.1.1
bandit==1.7.5
black==23.3.0
boltons==23.0.0
bytecode==0.14.1
CacheControl==0.12.11
cachetools==5.3.0
cattrs==22.2.0
certifi==2022.12.7
cffi==1.15.1
chardet==5.1.0
charset-normalizer==3.1.0
click==8.1.3
colorama==0.4.6
coverage==7.2.3
cryptography==40.0.1
ddsketch==2.0.4
ddtrace==1.13.0
Deprecated==1.2.13
deptry==0.8.0
diff-cover==7.5.0
dill==0.3.6
dnspython==2.3.0
email-validator==1.3.1
envier==0.4.0
eradicate==2.2.0
face==20.1.1
factory-boy==3.2.1
Faker==18.7.0
fastapi==0.95.1
fastapi-restful==0.4.3
firebase-admin==6.1.0
flake8==6.0.0
flake8-bugbear==23.5.9
flake8-builtins==2.1.0
flake8-comprehensions==3.12.0
flake8-debugger==4.1.2
flake8-docstrings==1.7.0
flake8-isort==6.0.0
flake8-noqa==1.3.1
flake8-print==5.0.0
Flake8-pyproject==1.2.3
flake8-quotes==3.3.2
flake8-string-format==0.3.0
flake8_simplify==0.20.0
GeoAlchemy2==0.13.1
gitdb==4.0.10
GitPython==3.1.31
glom==23.3.0
google-api-core==2.11.0
google-api-python-client==2.83.0
google-auth==2.17.1
google-auth-httplib2==0.1.0
google-cloud-core==2.3.2
google-cloud-firestore==2.11.0
google-cloud-storage==2.8.0
google-crc32c==1.5.0
google-resumable-media==2.4.1
googleapis-common-protos==1.59.0
grpcio==1.53.0
grpcio-status==1.53.0
h11==0.14.0
httpcore==0.16.3
httplib2==0.22.0
httptools==0.5.0
httpx==0.23.3
idna==3.4
importlib-metadata==6.0.1
iniconfig==2.0.0
isort==5.12.0
itsdangerous==2.1.2
Jinja2==3.1.2
jsonschema==4.17.3
lazy-object-proxy==1.9.0
markdown-it-py==2.2.0
MarkupSafe==2.1.2
mccabe==0.7.0
mdurl==0.1.2
more-itertools==9.1.0
msgpack==1.0.5
mypy-extensions==1.0.0
numpy==1.24.2
opentelemetry-api==1.17.0
opentelemetry-instrumentation==0.31b0
opentelemetry-instrumentation-asgi==0.31b0
opentelemetry-instrumentation-fastapi==0.31b0
opentelemetry-semantic-conventions==0.31b0
opentelemetry-util-http==0.31b0
orjson==3.8.9
packaging==23.0
pandas==2.0.1
pandas-vet==0.2.3
pathspec==0.11.1
pbr==5.11.1
pendulum==2.1.2
pep8-naming==0.13.3
platformdirs==3.2.0
pluggy==1.0.0
pprintpp==0.4.0
proto-plus==1.22.2
protobuf==4.22.1
psutil==5.9.4
psycopg2-binary==2.9.6
pyasn1==0.4.8
pyasn1-modules==0.2.8
pycodestyle==2.10.0
pycparser==2.21
pydantic==1.10.7
pydash==7.0.3
pydocstyle==6.3.0
pyflakes==3.0.1
Pygments==2.14.0
PyJWT==2.6.0
pylint==2.17.4
pylint-plugin-utils==0.7
pylint-pydantic==0.1.8
pylint-pytest==1.1.2
pyparsing==3.0.9
pyrsistent==0.19.3
pytest==7.3.1
pytest-clarity==1.0.1
pytest-cov==4.0.0
pytest-docker==1.0.1
pytest-mock==3.10.0
pytest-rerunfailures==11.1.2
pytest-sugar==0.9.7
pytest-testmon==2.0.6
python-dateutil==2.8.2
python-dotenv==1.0.0
python-multipart==0.0.6
pytz==2023.3
pytzdata==2020.1
pyupgrade==3.4.0
PyYAML==6.0
regex==2023.3.23
requests==2.28.2
rfc3986==1.5.0
rich==13.3.3
rsa==4.9
six==1.16.0
smmap==5.0.0
sniffio==1.3.0
snowballstemmer==2.2.0
SQLAlchemy==1.4.48
SQLAlchemy-Utils==0.41.0
sqlfluff==2.1.0
starlette==0.26.1
stevedore==5.0.0
tblib==1.7.0
tenacity==8.2.2
termcolor==2.2.0
time-machine==2.9.0
tokenize-rt==5.0.0
tomlkit==0.11.7
tqdm==4.65.0
typing_extensions==4.5.0
tzdata==2023.3
ujson==5.7.0
uritemplate==4.1.1
urllib3==1.26.15
uvicorn==0.22.0
uvloop==0.17.0
watchfiles==0.19.0
websockets==11.0
wrapt==1.15.0
xmltodict==0.13.0
yamllint==1.31.0
yesqa==1.4.0
zipp==3.15.0
` ` `
</details>

### How can we reproduce your problem?
Update the library and try query with sqlalchemy

### What is the result that you get?
The error above

### What is the result that you expected?
Not to error.

**Comment by brettlangdon** [comment]
@ollz272 which version did you upgrade from? (is the problem isolated to only 1.13.0?)

**Comment by ollz272** [comment]
@brettlangdon 1.12.7 -> 1.13.0

**Comment by mabdinur** [comment]
Hi @ollz272, 

Thanks for bringing this issue to our attention. Are you using psycopg3 in your application? I suspect this PR introduced this regression: 
[feat(tracing): support psycopg3](https://github.com/DataDog/dd-trace-py/commit/cc399442093fdb31ba16241efd7f439628cba75d). 

Also can you try using ddtrace v1.13.0 with `DD_TRACE_PSYCOPG_ENABLED=False`? This will help us isolate this issue to the psycopg integration.

**Comment by ollz272** [comment]
Hi @mabdinur, no we're still on psycopg2.

I'll give this a go tomorrow morning for you. I'll also try a small reproduction case.

**Comment by ollz272** [comment]
Hi @mabdinur 
This env var worked, thanks. I won't have time to make a small reproduction case unfortunately.

**Comment by mabdinur** [comment]
This issue should be fixed in the following release: https://github.com/DataDog/dd-trace-py/releases/tag/v1.13.3. 

Let me know if you're still seeing errors with the psycopg3 integration 😄

**Comment by ollz272** [comment]
@mabdinur We upgrade to 1.13.3 and we're still seeing issues. Just to confirm again, we are using psycopg2, not psycopg3.

**Comment by mabdinur** [comment]
Sorry for the confusion. I meant to say this issue should be resolved in 1.13.4. We plan to cut this release by the end of the week. Here's the latest draft: https://github.com/DataDog/dd-trace-py/releases/tag/untagged-3c440a2a20f967c81312

**Comment by mabdinur** [comment]
Hey @ollz272, 

Sorry I linked the wrong bug fix.

This issue appears to be duplicate of: https://github.com/DataDog/dd-trace-py/issues/5926. 

This issue should be resolved by the following [PR](https://github.com/DataDog/dd-trace-py/pull/5927). We will release this fix in ddtrace v1.13.4. Sorry about the confusion.

**Comment by emmettbutler** [comment]
1.13.4 has been released.
